src/main/java/org/sonarlint/intellij/config/global/SonarQubeServer.java

```diff
@@ -0,0 +1,150 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.config.global;
+
+import com.intellij.openapi.util.Comparing;
+import com.intellij.openapi.util.PasswordUtil;
+import com.intellij.util.xmlb.annotations.Tag;
+import com.intellij.util.xmlb.annotations.Transient;
+
+public class SonarQubeServer {
+  private String hostUrl;
+  private String token;
+
+  private String name;
+  private String login;
+  private String password;
+  private String serverId;
+  private boolean enableProxy;
+
+  public SonarQubeServer() {
+
+  }
+
+  public SonarQubeServer(SonarQubeServer toCopy) {
+    setHostUrl(toCopy.getHostUrl());
+    setServerId(toCopy.getServerId());
+    setPassword(toCopy.getPassword());
+    setToken(toCopy.getToken());
+    setLogin(toCopy.getLogin());
+    setName(toCopy.getName());
+    setEnableProxy(toCopy.enableProxy());
+  }
+
+  @Override
+  public boolean equals(Object o){
+    if (!(o instanceof SonarQubeServer)) return false;
+    SonarQubeServer other = (SonarQubeServer)o;
+
+    return Comparing.equal(getHostUrl(), other.getHostUrl()) &&
+      Comparing.equal(getServerId(), other.getServerId()) &&
+      Comparing.equal(getPassword(), other.getPassword()) &&
+      Comparing.equal(getToken(), other.getToken()) &&
+      Comparing.equal(getLogin(), other.getLogin()) &&
+      Comparing.equal(getName(), other.getName()) &&
+      Comparing.equal(enableProxy(), other.enableProxy());
+  }
+
+  public String getLogin() {
+    return login;
+  }
+
+  public void setLogin(String login) {
+    this.login = login;
+  }
+
+  public String getHostUrl() {
+    return hostUrl;
+  }
+
+  public void setHostUrl(String hostUrl) {
+    this.hostUrl = hostUrl;
+  }
+
+  @Transient
+  public String getToken() {
+    return token;
+  }
+
+  @Tag("token")
+  public String getEncodedToken() {
+    return PasswordUtil.encodePassword(getToken());
+  }
+
+  public void setEncodedToken(String token) {
+    try {
+      setToken(PasswordUtil.decodePassword(token));
+    }
+    catch (NumberFormatException e) {
+      // do nothing
+    }
+  }
+
+  public boolean enableProxy() {
+    return enableProxy;
+  }
+
+  public void setEnableProxy(boolean enableProxy) {
+    this.enableProxy = enableProxy;
+  }
+
+  @Transient
+  public String getPassword() {
+    return password;
+  }
+
+  @Tag("password")
+  public String getEncodedPassword() {
+    return PasswordUtil.encodePassword(getPassword());
+  }
+
+  public void setEncodedPassword(String password) {
+    try {
+      setPassword(PasswordUtil.decodePassword(password));
+    }
+    catch (NumberFormatException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this "catch" on the same line that the previous closing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)